### PR TITLE
Unify CI workflows

### PR DESCRIPTION
**Pull request by michalinacienciala**
We have CI workflows scattered across various repositories. We try to keep their implementation as similar as possible. In this PR and the ones listed below we introduce a couple of changes which unify used solutions and fix small mistakes, such as:
* making tagging of published packages consistent
* passing `NPM_TOKEN` secret as variable instead of passing it by writing to file
* handling caching of npm/yarn dependencies via `setup-node` action
* adding `workflow_dispatch` job trigger where it's missing
* using Node.js in version 14.x instead of 12.x where possible

Related PRs in other repositories:
* https://github.com/keep-network/keep-core/pull/2776
* https://github.com/keep-network/keep-ecdsa/pull/941
* https://github.com/keep-network/tbtc/pull/842
* https://github.com/keep-network/tbtc.js/pull/145
* https://github.com/keep-network/sortition-pools/pull/145
* https://github.com/keep-network/local-setup/pull/124
* https://github.com/keep-network/coverage-pools/pull/204
* https://github.com/threshold-network/solidity-contracts/pull/59

### CODE REVIEW

File: .github/workflows/yearn.yml [hunk 1]
```diff
@@ -42,6 +43,8 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: yarn/yarn.lock
```
Comment (nkuba) [inline .github/workflows/yearn.yml:47]:
```suggestion
          cache-dependency-path: yearn/yarn.lock
```

File: .github/workflows/yearn.yml [hunk 2]
```diff
@@ -69,6 +72,8 @@ jobs:
...
```
Comment (nkuba) [inline .github/workflows/yearn.yml:76]:
```suggestion
          cache-dependency-path: yearn/yarn.lock
```

**Review by nkuba — CHANGES_REQUESTED** [review]

File: .github/workflows/yearn.yml (hunk 1 above)
Comment (michalinacienciala) [reply .github/workflows/yearn.yml]: Good catch, fixed in 108ceff.

File: .github/workflows/yearn.yml (hunk 2 above)
Comment (michalinacienciala) [reply .github/workflows/yearn.yml]: Good catch, fixed in 108ceff.

**Review by nkuba — APPROVED** [review]

**Pull request merged**